exercises/src/main/scala/fpinscala/monads/Monad.scala

```diff
@@ -34,25 +34,49 @@ trait Monad[M[_]] extends Functor[M] {
   def map2[A,B,C](ma: M[A], mb: M[B])(f: (A, B) => C): M[C] =
     flatMap(ma)(a => map(mb)(b => f(a, b)))
 
-  def sequence[A](lma: List[M[A]]): M[List[A]] = ???
-
-  def traverse[A,B](la: List[A])(f: A => M[B]): M[List[B]] = ???
-
-  def replicateM[A](n: Int, ma: M[A]): M[List[A]] = ???
-
-  def compose[A,B,C](f: A => M[B], g: B => M[C]): A => M[C] = ???
+  def sequence[A](lma: List[M[A]]): M[List[A]] =
+    lma.foldRight(unit(Nil:List[A]))((p, acc) => {
```

Comment:
It's mostly about style, but personally I prefer to use `List.empty[A]` to have an empty list with the appropriate type.